Assets/Scripts/Managers/SoundManager.cs

```diff
@@ -58,9 +58,16 @@ void Start()
             m_AudioSource.clip = m_AudioClips[1];
             m_AudioSource.Play();
         }
-    }
-
+        DontDestroyOnLoad(this);
 
+    }
+    void Update()
+    {
+        if (SceneManager.GetActiveScene().name == "GameSceneFinal")
+        {
+            m_AudioSource.Stop();
+        }
```

Comment:
this deserves a comment at least to explain, from my view, it looks like you are stopping the audio source every frame inside of the main game scene? Why?